packages/api/src/Application/Adapters/Schemas/UserSchema.ts

```diff
@@ -0,0 +1,119 @@
+import Joi from '@hapi/joi';
+
+export const UserCreateSchema = Joi.object()
+  .keys({
+    name: Joi.string()
+      .min(2)
+      .max(30)
+      .required(),
+    lastname: Joi.string()
+      .min(2)
+      .max(30)
+      .required(),
+    age: Joi.number()
+      .min(1)
+      .max(110),
+    dni: Joi.number()
+      .min(1000000)
+      .max(70000000),
+    email: Joi.string()
+      .email({ minDomainSegments: 2, tlds: { allow: ['com', 'net'] } })
+      .required(),
+    birthyear: Joi.number()
+      .min(1900)
+      .max(2150),
+    password: Joi.string()
+      .pattern(new RegExp('^[a-zA-Z0-9]{8,30}$'))
+      .required(),
+    confirmation: Joi.ref('password'),
+    phonenumber: Joi.number()
+      .min(6)
+      .max(12),
+    cellphonenumber: Joi.number()
+      .min(8)
+      .max(15),
+    phoneareacode: Joi.number()
+      .min(2)
+      .max(8),
+    city: Joi.string()
+      .min(4)
+      .max(20),
+    state: Joi.string()
+      .min(4)
+      .max(30),
+    country: Joi.string()
+      .min(3)
+      .max(20),
+  })
+  .with('password', 'confirmation');
+
+export const UserEditSchema = Joi.object()
+  .keys({
+    id: Joi.number()
+      .min(0)
+      .required(),
+    name: Joi.string()
+      .min(2)
+      .max(30)
+      .required(),
+    lastname: Joi.string()
+      .min(2)
+      .max(30)
+      .required(),
+    age: Joi.number()
+      .min(1)
+      .max(110)
+      .required(),
+    dni: Joi.number()
+      .min(1000000)
+      .max(70000000)
+      .required(),
+    email: Joi.string()
+      .email({ minDomainSegments: 2, tlds: { allow: ['com', 'net'] } })
+      .required(),
+    birthyear: Joi.number()
+      .min(1900)
+      .max(2150)
+      .required(),
+    password: Joi.string()
+      .pattern(new RegExp('^[a-zA-Z0-9]{8,30}$'))
+      .required(),
+    confirmation: Joi.ref('password'),
```

Comment:
Aquí password and confirmation no irian ya que solamente es para editar campos del perfil

Reply:
Pero y si el usuario quiere editar su contraseña como se haría?? No se hace por acá??

Reply:
Se haría en otro endpoint más simple